### DISCUSSION TITLE
Redraw with new positions

### DISCUSSION THREAD

**Discussion by jkanel** [open-source-labs/Svelvet#358] [Q&A] [answered]
I have been using **ngraph.forceLayout** to calculate the layout of the `<Svelvet >` diagram.   Works great!  And I can allow users to show / hide `<Node>` elements based on setting filters.  So far so good.

Now, I'd like to be able to show / hide `<Node>` elements  and then have the layout recalculate with only the remaining visible `<Node>` elements.  This implies that `<Node>` positions will change.  However, Svelvet will not re-render the nodes after positions are updated.

Is there a way to force a redraw of `<Node>` positions without a page refresh?

**Comment by briangregoryholmes** [comment, marked as answer]
In a custom Node, you can access the underlying store object via the following code in a script tag.

```js
const graph = getContext('graph');
let internalNode;

onMount(() => {
    internalNode = graph.nodes.get(`N-${yourNodeID}`);
});
```

You could then update the position manually by calling `internalNode.position.set({x, y})`.

This isn't really for external use and I don't recommend building something around this functionality as it is subject to change. The next version, which will release in a few weeks, will have exported functions and stores on the major components that you will be able to access via component instance binding. This is how we plan on enabling this kind of functionality going forward and we will have updated documentation for exactly this use case.

You can also access this internal Node object via `let:node` on the Node component itself, but it would not be accessible in the script tag.

**Reply by jkanel** [reply]
Thanks @briangregoryholmes.  If it's a few weeks, I'm tempted to wait!
